Purchases/Public/Purchases.swift

```diff
@@ -732,28 +702,31 @@ public extension Purchases {
      * If this method is called and the current user is anonymous, it will return an error.
      * See https://docs.revenuecat.com/docs/user-ids
      */
-    @objc func logOut(completionBlock maybeCompletion: ReceivePurchaserInfoBlock?) {
+    @objc(logOutWithCompletion:)
+    func logOut(_ completion: ((PurchaserInfo?, Error?) -> Void)?) {
         identityManager.logOut { maybeError in
             guard maybeError == nil else {
-                if let completion = maybeCompletion {
+                if let completion = completion {
                     self.operationDispatcher.dispatchOnMainThread {
                         completion(nil, maybeError)
                     }
                 }
                 return
             }
 
-            self.updateAllCaches(completion: maybeCompletion)
+            self.updateAllCaches(completion: completion)
         }
     }
 
     /**
      * Resets the Purchases client clearing the saved appUserID.
      * This will generate a random user id and save it in the cache.
      */
-    @objc func reset(completionBlock maybeCompletion: ReceivePurchaserInfoBlock?) {
+    @available(*, deprecated, message: "use logOut instead", renamed: "logOut")
+    @objc(resetWithCompletion:)
+    func reset(_ completion: ((PurchaserInfo?, Error?) -> Void)?) {
```

Comment:
not sure the `_`s are a good idea - folks can already skip the completion block by just doing 
```swift
reset { purchaserInfo, error in
    // 
}
```

So I think it'd be fine to keep the block names, and if we did, we'd also get rid of the `@objc` names... which means that we're providing the exact same interface for obj-c and swift, which is a win I think

Reply:
💯

Reply:
I just made a few updates for this across our public API 😄